File: SRC.NET/BCVariable.cs

namespace Project1
{
    static class BCVariable
    {

        // Copyright (C) 1997-2012 Kei Sakamoto / Inui Tetsuyuki
        // 本プログラムはフリーソフトであり、無保証です。
        // 本プログラムはGNU General Public License(Ver.3またはそれ以降)が定める条件の下で
        // 再頒布または改変することができます。

        // バトルコンフィグデータが有効か？
        public static bool IsConfig;

        // バトルコンフィグデータの各種変数を定義する

        // バトルコンフィグデータ対象中心ユニット定義
        // ---------かならず定義されるデータ
        public static Unit MeUnit;

        // 攻撃側ユニット定義
        public static Unit AtkUnit;

        // 防御側ユニット定義
        public static Unit DefUnit;

        // 武器番号
        public static short WeaponNumber;

        // ---------定義されない場合もある(計算後にリセットされる)データ
        // 攻撃値
        public static int AttackExp;

        // 攻撃側定義変数
        public static int AttackVariable;

        // 防御側定義変数
        public static int DffenceVariable;

        // 地形補正
        public static double TerrainAdaption;

        // サイズ補正
        public static double SizeMod;

        // 最終値
        public static int LastVariable;

        // 武器攻撃力
        public static int WeaponPower;

        // 装甲値
        public static int Armor;

        // ザコ補正
        public static int CommonEnemy;

        // 定義されないこともあるデータをここでリセットする
        public static void DataReset()
        {
            AttackExp = 0;
            AttackVariable = 0;
            DffenceVariable = 0;
            TerrainAdaption = 1d;
            SizeMod = 1d;
            LastVariable = 0;
            WeaponPower = 0;
            CommonEnemy = 0;
        }
    }
}

File: SRC.NET/BCVariable.vb
Option Strict Off
Option Explicit On
Module BCVariable
	
	' Copyright (C) 1997-2012 Kei Sakamoto / Inui Tetsuyuki
	'Invalid_string_refer_to_original_code
	'Invalid_string_refer_to_original_code
	'Invalid_string_refer_to_original_code
	
	'Invalid_string_refer_to_original_code
	Public IsConfig As Boolean
	
	'Invalid_string_refer_to_original_code
	
	'Invalid_string_refer_to_original_code
	' ---------かならず定義されるデータ
	Public MeUnit As Unit
	
	'Invalid_string_refer_to_original_code
	Public AtkUnit As Unit
	
	' 防御側ユニット定義
	Public DefUnit As Unit
	
	' 武器番号
	Public WeaponNumber As Short
	
	'Invalid_string_refer_to_original_code
	'Invalid_string_refer_to_original_code
	Public AttackExp As Integer
	
	'Invalid_string_refer_to_original_code
	Public AttackVariable As Integer
	
	' 防御側定義変数
	Public DffenceVariable As Integer
	
	' 地形補正
	Public TerrainAdaption As Double
	
	' サイズ補正
	Public SizeMod As Double
	
	' 最終値
	Public LastVariable As Integer
	
	'Invalid_string_refer_to_original_code
	Public WeaponPower As Integer
	
	'Invalid_string_refer_to_original_code
	Public Armor As Integer
	
	' ザコ補正
	Public CommonEnemy As Integer
	
	'Invalid_string_refer_to_original_code
	Public Sub DataReset()
		AttackExp = 0
		AttackVariable = 0
		DffenceVariable = 0
		TerrainAdaption = 1
		SizeMod = 1
		LastVariable = 0
		WeaponPower = 0
		CommonEnemy = 0
	End Sub
End Module